compiler/optimizer/FieldPrivatizer.hpp

```diff
@@ -163,6 +173,20 @@ class TR_FieldPrivatizer : public TR_LoopTransformer
    List<TR::TreeTop> _appendCalls;
 
    TR_PostDominators *_postDominators;
+
+   /**
+    * Tracks whether a node and its subtrees have been checked to
+    * see whether they contain a \ref TR::instanceof operation.
+    * If so, \ref _subtreeHasInstanceOf can be checked for the
+    * presence of an \c TR::instanceof in the subtree.
+    */
+   TR::NodeChecklist _subtreeCheckedForInstanceOf;
+
+   /**
+    * Tracks whether a node or one or its subtrees performs a
+    * \ref TR::instanceof operation.
+    */
+   TR::NodeChecklist _subtreeHasInstanceOf;
```

Comment:
These fields are only initialized when the optimization is first created.  At the moment I believe field privatization runs only once.  But to future-proof this opt can you reinitialize the `_subtreeHasInstanceOf` and `_subtreeCheckedForInstanceOf` fields at the beginning of each invocation of this opt (e.g., in the `perform()` function)?

Reply:
Opt pass objects are newly created for each pass:
https://github.com/eclipse/omr/blob/52099bb355a055fbfd3a517b37f7cfc64c4546fe/compiler/optimizer/OMROptimizer.cpp#L1686-L1693
The `delete` there and further below (after calling `perform()`) suggest that the factory creates a fresh instance with `new`, which is what I thought was the case. I looked at a number of `OptimizationManager` factory functions as a quick check on that. These ones:
https://github.com/eclipse/omr/blob/52099bb355a055fbfd3a517b37f7cfc64c4546fe/compiler/optimizer/OMROptimizer.cpp#L819-L844
And indeed they all return a `new` instance. In particular, that includes `TR_FieldPrivatizer::create`:
https://github.com/eclipse/omr/blob/52099bb355a055fbfd3a517b37f7cfc64c4546fe/compiler/optimizer/FieldPrivatizer.hpp#L122-L125)

Reply:
Daryl @0xdaryl, would you still prefer that I future-proof this against the possibility of `perform` being called multiple times for an instance of the optimization?

Reply:
No, that's fine.  Thanks for the detailed explanation @jdmpapin .